pkg/image/registry/image/etcd/etcd.go

```diff
@@ -59,8 +59,27 @@ func NewREST(optsGetter restoptions.Getter) (*REST, error) {
 	}
 
 	if err := restoptions.ApplyOptions(optsGetter, store, prefix); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return &REST{store}, nil
+	internalStore := *store
+	internalStore.UpdateStrategy = image.InternalStrategy
+	internalStore.CreateStrategy = nil
+
+	return &REST{store}, &InternalREST{store: &internalStore}, nil
+}
+
+// InternalREST implements a RESTStorage for image resources internally modifiable.
```

Comment:
It allows to modify image from internal controllers. Since image is mostly immutable, currently it allows to modify only image signatures. It is now used in image signature rest and will be utilized also once we are able to parse signatures and fill/update their metadata.